File: 2020/day1/day1.go
package main

import (
	"fmt"
	"github.com/kamillo/aoc/utils"
	"strconv"
)

func main() {
	lines := utils.GetLines("input.txt")
	first, second := 0.0, 0.0

	numbers := make([]float64, len(lines))
	for i, line := range lines {
		numbers[i], _ = strconv.ParseFloat(line, 64)
	}

	for i, number1 := range numbers {
		for j, number2 := range numbers {
			if i != j && first == 0 {
				if number1+number2 == 2020 {
					first = number1 * number2
				}
			}

			for k, number3 := range numbers {
				if i != j && j != k && i != k {
					if number1+number2+number3 == 2020 {
						second = number1 * number2 * number3
						break
					}
				}
			}
			if second != 0 && first != 0 {
				break
			}
		}
		if second != 0 && first != 0 {
			break
		}
	}
	fmt.Printf("First part: %f\n", first)
	fmt.Printf("Second part: %f\n", second)
}

File: nim/2020/day1/day1.nim
from strutils import parseInt
import utils/fileutils

var 
    first = 0
    second = 0
    lines = getLines("input.txt")
    numbers: seq[int]

for line in lines:
    numbers.add(parseInt(line))

for i, number1 in numbers:
    for j, number2 in numbers:
        if i != j and first == 0 and number1+number2 == 2020:
            first = number1 * number2        
        for k, number3 in numbers:
            if i != j and j != k and i != k and number1 + number2 + number3 == 2020:
                second = number1 * number2 * number3
                break
        if second != 0 and first != 0:
            break
    if second != 0 and first != 0:
        break
        
echo("Part 1: ", first)
echo("Part 2: ", second)